README.md

```diff
@@ -124,6 +124,45 @@ CompletionStage<String> f2;
 CompletionStage<String> result = dereference(combine(combined -> completedFuture(combined.get(f1) + combined.get(f2)), f1, f2));
 ```
 
+### Unwrapping CompletionExceptions
+
+When handling exceptions thrown by `join()` and `get()` or received in `whenComplete()`, 
+`exceptionally()` or `handle()`, it can be difficult to figure out if the exception that is 
+obtained is going to be a `CompletionException` or some other `Throwable`. Even when we know
+that the exception is guaranteed to be a `CompletionException`, it can be hard to know ahead
+of time if `ex.getCause()` is going to be another `CompletionException`, an actually useful
+`Throwable`, or deeply nested the `CompletionExceptions` might be.
+
+`unravel(Throwable)` will recursively unwrap `CompletionExceptions` at the top of the exception
+chain. It can be used this way:
+
+```java
+try {
+    String value = future.join();
```

Comment:
This looks useful! Maybe there could also be a utility function handling the joining on the future itself/? So:

```java
try {
  // Calls `future.toCompletableFuture().join()` and unwraps `CompletionException`s
  String value = joinUnraveled(future);
  // (Could define equivalents for `get()` too)
} catch (BusinessRelevantException e) {
  // ...
}
```

Reply:
Yeah, I was thinking the same. If you think this looks useful, then I guess I'll put a bit more effort in it. 

I haven't entirely convinced myself that `unravel` was the perfect name for this method, but it seemed better than `unwrap` (which is already used, and this function only unwraps in certain cases), or `extractCause`. 

Do you maybe have any better suggestion for the name?

Reply:
For what it's worth I call these `unwrap` but I like [unravel](https://www.ea.com/games/unravel). Plus it's from a Swedish game studio so that settles it.